src/plugins/instrument_fetch.js

```diff
@@ -158,42 +158,50 @@ class InstrumentFetch {
 
         return function (request, options = {}) {
             request = new Request(request, options);
-            const url = request.url;
             const opts = tracer.options();
 
-            if (!self._shouldTrace(tracer, url)) {
+            if (!self._shouldTrace(tracer, request.url)) {
                 return proxiedFetch.apply(null, arguments);
             }
 
             let span = tracer.startSpan('fetch');
             tracer.addActiveRootSpan(span);
 
+            const parsed = new URL(request.url);
             let tags = {
-                method : options && options.method ? options.method : 'GET',
-                url    : url,
-            };
-            if (url) {
-                tags.url_pathname = url.split('?')[0];
-            }
-
-            const fetchPayload = Object.assign({}, tags);
+                method : request.method,
+                url    : request.url,
 
+                // NOTE: these are potentially sensitive values, but if a
+                // consumer doesn't want them traced, they can exclude URLs
+                // matching this pattern.
+                username : parsed.username,
+                password : parsed.password,
```

Comment:
is there ever a time we'd like to trace username/password? or I assume everything is just included for completeness?

Reply:
nvm, I just saw your earlier comment addressing this.

Reply:
I was thinking that we probably shouldn't do this by default (I can imagine this getting added on accidentally and causing PII problems)

Reply:
I wonder if we should also guard against it by checking if the url returns a `password` or `username`. Then we could default for anything that does to mask said url in log

Reply:
I just included everything for completeness. It's definitely rare, but... why not?

Reply:
Oh, by mask did you mean log a URL like `https://xxxxxx:xxxxxx@/rest/of/url`?

Reply:
Oh, by mask did you mean log a URL like `https://xxxxxx:xxxxxx@/rest/of/url`?

Reply:
So, I vote for B (and is what's currently pushed). Seems like C is the right solution long term, but that it probably has more repercussions for the lightstep service itself. I'll leave it for the project to decide / implement / not implement on a subsequent PR.

Reply:
B sounds good. we won't actively log usernames and passwords. but the URL itself is left intact. then it's up to the consumer to blacklist it, just like before this change.